stdlib/proposals/ai/evaluation_for_ai_agents_in_ballerina.md

```diff
@@ -0,0 +1,321 @@
+
+- Monitoring performance trends over time
+- Comparing current results against historical baselines
+- Querying and analyzing metrics across versions, tasks, or datasets
+
+While the Ballerina testing framework can replicate the **evaluation logic**, it does **not** provide built-in support for persistent storage, trend analysis, or historical comparisons. These capabilities could be added by exporting evaluation results to an external storage system—and in the future, they could be offered through **Devant** as a first-class feature.
+
+For most integration and deployment scenarios, however, the proposed test-driven evaluation approach is more than sufficient to ensure agent reliability and prevent regressions.
+
+Developers who prefer observability-based evaluation can still integrate existing tooling. As long as the agent is configured with a compatible observability provider, frameworks like Arize Phoenix can be used without friction. 
+
+For example:
+1. Configure Arize as the observability backend so that agent traces are exported directly to Arize.
+2. Use Arize’s REST API to run evaluations and submit results back to the Arize platform.
+
+This approach ensures that both **trace-free** and **observability-driven** evaluation workflows remain fully supported.
+
+### Further enhancement to the Ballerina test framework
+
+To overcome the limitation around storing and tracking past evaluation outputs, the Ballerina test framework can be extended with a set of improvements that make evaluation results easier to retain, organize, and analyze over time.
+
+* Support a dedicated directory for storing evaluation reports by using `bal test --test-report --test-report-dir evaluation`. This places all generated reports inside an `evaluation` folder, allowing teams to version and track them through source control.
+* Produce a new report file for every test execution that includes a timestamp in its name. This ensures older reports remain intact, making it possible to review historical data and understand how performance changes across versions.
+* Include the input values passed to tests through the `dataProvider` field in the annotations. Recording this information makes each report self-contained so developers can fully understand what was evaluated without checking the test sources.
+
+With these additions, the Ballerina Test framework can serve both immediate regression checks and longer term evaluation tracking.
+
+## Summary of changes required
+
+### Changes to the `ballerina/ai` module
+
+1. **New Record Types**
+   - Add `Trace` record to represent agent execution traces
+   - Add `Iteration` record to represent individual steps in execution
+   - Add `ToolSchema` record to capture tool metadata
+
+2. **Agent API Enhancement**
+   - Modify `Agent.run()` method to support type binding with `typedesc<Trace|string>` parameter
+   - Return either string response or full `Trace` object based on the type descriptor
```

Comment:
record or object?

An object could also work, if the util functions can be methods.